### PR TITLE
Migrate away from google.com gcp project k8s-testimages

### PR DESCRIPTION
**Pull request by cpanato**
**What this PR does / why we need it**:

- Migrate away from google.com gcp project k8s-testimages

Related to https://github.com/kubernetes/k8s.io/issues/1523

/assign @ameukam @micahhausler

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -1,15 +1,15 @@
 options:
   substitution_option: ALLOW_LOOSE
 steps:
-  - name: gcr.io/k8s-testimages/gcb-docker-gcloud
+  - name: gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20211118-2f2d816b90
```
Comment (dims) [inline cloudbuild.yaml:4]: @cpanato looks like newer ones are available? https://cs.github.com/?scopeName=All+repos&scope=&q=gcr.io%2Fk8s-staging-test-infra%2Fgcb-docker-gcloud%3Av2022+org%3Akubernetes 

can you please update?

File: cloudbuild.yaml [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
     entrypoint: /buildx-entrypoint
     args:
-    - build
-    - --tag=gcr.io/$PROJECT_ID/aws-iam-authenticator:$_GIT_TAG
-    - --tag=gcr.io/$PROJECT_ID/aws-iam-authenticator:latest
-    - --platform=linux/amd64,linux/arm64
-    - --output=type=registry
-    - .
+      - build
```
Comment (dims) [inline cloudbuild.yaml:7]: Do we need these whitespace changes?

File: cloudbuild.yaml (hunk 1 above)
Comment (cpanato) [reply cloudbuild.yaml:4]: done, ptal
thanks for the catch

File: cloudbuild.yaml (hunk 2 above)
Comment (cpanato) [reply cloudbuild.yaml:7]: there no whitespace it is just indentation for better yaml format

**Pull request merged**